src/value/use-velocity.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from "react"
+import { MotionValue } from "."
+import { useMotionValue } from "./use-motion-value"
+/**
+ * Creates a `MotionValue` that updates when the velocity of the provided `MotionValue` changes.
+ *
+ * ```javascript
+ * const x = useMotionValue(0)
+ * const xVelocity = useVelocity(x)
+ * const xAcceleration = useVelocity(xVelocity)
+ * ```
+ *
+ * @public
+ */
+export function useVelocity(value: MotionValue<number>): MotionValue<number> {
+    const velocity = useMotionValue(value.getVelocity())
+
+    useEffect(() => {
+        return value.velocityUpdateSubscribers.add((newVelocity) => {
+            velocity.set(newVelocity)
```

Comment:
We could theoretically just subscribe to the value and then call `getVelocity` on that, right?

Reply:
That was my preferred solution but there is a caveat that led me to putting it inside `MotionValue`. Every time a `MotionValue` is updated it schedules a velocity check at the end of the subsequent frame, so if it isn't updated in the meantime we will correctly report the velocity as `0` if requested.

It's this "if requested"  bit that's the problem. We'd have to duplicate that kind of scheduling here if we were to make sure the velocity is updated correctly.  By making explicit velocity subscribers we can ensure they get called at the same time velocity is updated, no matter the exact mechanism we use to perform that now or in the future.

Reply:
Makes sense 👍🏻